### PR TITLE
Provide initial rook artifacts

### PR DESCRIPTION
**Pull request by karmab**

### CODE REVIEW

File: manifests/131_rook_cm.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+apiVersion: v1
+data:
+  csi-cephfsplugin-provisioner.yaml: "kind: StatefulSet\napiVersion: apps/v1beta2\nmetadata:\n
```
Comment (vrutkovs) [inline manifests/131_rook_cm.yaml]:
```suggestion
  csi-cephfsplugin-provisioner.yaml: |
    kind: StatefulSet
    apiVersion: apps/v1beta2
    metadata:
```
etc.

**Comment by karmab** [comment]
@vrutkovs i didn't came up with the manifests, mostly used the existing ones from rook so i believe they actually should be let untouched

**Comment by vrutkovs** [comment]
@karmab if these are upstream why not `curl` them? If these are expected to be changed these should be readable

**Comment by karmab** [comment]
i m fine providing a 09 rook script, actually i prefer it
@hardys happy with that approach ?

**Comment by karmab** [comment]
updated the PR to use a dedicated 10 script

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/310/

**Comment by karmab** [comment]
unreladed @derekhiggins ?

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/314/

**Comment by hardys** [comment]
I'm fine with the separate scripts, but this isn't enabling any extra disks for the VM setup - I guess we need those for the OSD's, so can you clarify how many and what size?

The ansible roles already have the templating to add the disks, so it should be pretty easy to add, even if we do it as a follow-up

File: 10_deploy_rook.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/bash
+
```
Comment (hardys) [inline 10_deploy_rook.sh]: We need to source common.sh here so the KUBECONFIG variable is set, ref https://github.com/openshift-metalkube/dev-scripts/pull/177/

Comment (hardys) [reply 10_deploy_rook.sh]: Also probably set -eux similar to other scripts?

File: 10_deploy_rook.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+git clone https://github.com/rook/rook
+cd rook/cluster/examples/kubernetes/ceph
```
Comment (hardys) [inline 10_deploy_rook.sh]: This git clone should go in `03_ocp_repo_sync.sh` I think, so it goes in the GOPATH and not in the dev-scripts directory?

File: 10_deploy_rook.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+oc apply -f csi/rbac/cephfs
+sed -i '/ROOK_HOSTPATH_REQUIRES_PRIVILEGED/!b;n;c\          value: "true"' operator-with-csi.yaml
+oc create -f scc.yaml
+oc adm policy add-scc-to-user privileged -z rook-csi-rbd-provisioner-sa -n rook-ceph-system
+oc adm policy add-scc-to-user privileged -z rook-csi-rbd-attacher-sa -n rook-ceph-system
+oc adm policy add-scc-to-user privileged -z csi-rook-csi-rbd-plugin-sa -n rook-ceph-system
+oc adm policy add-scc-to-user privileged -z rook-csi-cephfs-plugin-sa -n rook-ceph-system
+oc adm policy add-scc-to-user privileged -z rook-csi-cephfs-provisioner-sa -n rook-ceph-system
+oc create -f operator-with-csi.yaml
+sleep 180
```
Comment (hardys) [inline 10_deploy_rook.sh]: Can we poll or -w for some status/event instead of sleeping here?

File: 10_deploy_rook.sh [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+metadata:
+  name: rbd
+  namespace: rook-ceph
+spec:
+  failureDomain: osd
+  replicated:
+    size: 3
+EOF
+
+sleep 240
```
Comment (hardys) [inline 10_deploy_rook.sh]: Same as above, for CI we want to wait for as little time as possible, and some poll/event check will be generally more robust?

See #130 for example where we started using oc wait to ensure a given condition or event happened?

File: 10_deploy_rook.sh [hunk 6]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/bash
+
+git clone https://github.com/rook/rook
```
Comment (ktdreyer) [inline 10_deploy_rook.sh]: We can do a shallow clone here to save time, right?

**Comment by ktdreyer** [comment]
I work on RH's Ceph team, and we have some similar bash code for running Rook with OpenShift. Is there any meeting where your team is discussing this approach?

This example YAML approach is changing frequently (eg. https://github.com/rook/rook/pull/2820) so I want to be sure we're aligned.

**Comment by hardys** [comment]
> I work on RH's Ceph team, and we have some similar bash code for running Rook with OpenShift. Is there any meeting where your team is discussing this approach?

Certainly happy to chat, but note the idea is that we consume whatever is the best-practice approach to deploy Ceph via Rook on OpenShift 4.x, not to reimplement anything that may already exist :)

These scripts are really just bootstrapping some temporary prototyping, and we expect/hope that ultimately we don't have any scripts and just some yaml manifests which are applied to the cluster via the machine-config-operator.

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/330/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/331/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/332/

File: 10_deploy_rook.sh (hunk 3 above)
Comment (karmab) [reply 10_deploy_rook.sh]: done

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/333/

**Comment by hardys** [comment]
I tested this and the nodes didn't deploy correctly - is this working locally for you?

I'm wondering if we need to change the ironic root device hint, as it's perhaps writing the disk image to the wrong drive, or booting from the wrong one?

I'll investigate further if you think this should be working as-is

**Comment by karmab** [comment]
@hardys i'm also hitting issues too with a secondary disk for the vms

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/342/

**Comment by karmab** [comment]
as a first pass, let s get the script available, we can investigate the additional disks failure separately

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/351/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/355/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/357/

**Comment by hardys** [comment]
Ok we discussed and agreed to merge this as-is, and work on adding the extradisks support as a followup, since this can be tested now by manually adding the disks.

I'll take a look at why the extradisks stuff doesn't work, we had it enabled when this repo was originally created and it worked, so hopefully it's a simple fix to ensure the deploy/boot happens on the correct disk.

**Pull request merged**